src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaHttpUtil.cs

```diff
@@ -0,0 +1,147 @@
+                    httpMethod = request.HttpMethod;
+                    var hostHeader = GetHeaderValue(request, HeaderHost);
+                    (hostName, hostPort) = GetHostAndPort(httpScheme, hostHeader);
+                    break;
+                case APIGatewayHttpApiV2ProxyRequest requestV2:
+                    httpScheme = GetHeaderValue(requestV2, HeaderXForwardedProto);
+                    httpTarget = string.Concat(requestV2?.RequestContext?.Http?.Path ?? string.Empty, ConstructQueryString(requestV2.QueryStringParameters));
+                    httpMethod = requestV2?.RequestContext?.Http?.Method;
+                    var hostHeaderV2 = GetHeaderValue(requestV2, HeaderHost);
+                    (hostName, hostPort) = GetHostAndPort(httpScheme, hostHeaderV2);
+                    break;
+            }
+
+            tags.AddTagIfNotNull(SemanticConventions.AttributeHttpScheme, httpScheme);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeHttpTarget, httpTarget);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeHttpMethod, httpMethod);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeNetHostName, hostName);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeNetHostPort, hostPort);
+
+            return tags;
+        }
+
+        internal static void SetHttpTagsFromResult(Activity activity, object result)
+        {
+            switch (result)
+            {
+                case APIGatewayProxyResponse response:
+                    activity.SetTag(SemanticConventions.AttributeHttpStatusCode, response.StatusCode.ToString());
+                    break;
+                case APIGatewayHttpApiV2ProxyResponse responseV2:
+                    activity.SetTag(SemanticConventions.AttributeHttpStatusCode, responseV2.StatusCode.ToString());
+                    break;
+            }
+        }
+
+        internal static (string Host, int? Port) GetHostAndPort(string httpScheme, string hostHeaders)
+        {
+            if (hostHeaders == null)
+            {
+                return (null, null);
+            }
+
+            // In case of multiple headres we consider only the 1st.
+            var hostHeader = hostHeaders.Split(',').First();
+            var hostAndPort = hostHeader.Split(new char[] { ':' }, 2);
+            if (hostAndPort.Length > 1)
+            {
+                var host = hostAndPort[0];
+                return int.TryParse(hostAndPort[1], out var port)
+                    ? (host, port)
+                    : (host, null);
+            }
+            else
+            {
+                int? defaultPort = null;
+                switch (httpScheme)
+                {
+                    case "http":
+                        defaultPort = 80;
+                        break;
+                    case "https":
+                        defaultPort = 443;
+                        break;
+                }
```

Comment:
```suggestion
                int? defaultPort = httpScheme == "https" ? 443 : httpScheme == "http" ? 80 : null;
```

Reply:
Introduced `GetDefaultPort` method and moved this simplification there.